pkg/sdn/plugin/subnets.go

```diff
@@ -95,11 +95,6 @@ func (master *OsdnMaster) deleteNode(nodeName string) error {
 	if err != nil {
 		return fmt.Errorf("Error fetching subnet for node %q for deletion: %v", nodeName, err)
 	}
-	_, ipnet, err := net.ParseCIDR(sub.Subnet)
-	if err != nil {
-		return fmt.Errorf("Error parsing subnet %q for node %q for deletion: %v", sub.Subnet, nodeName, err)
-	}
-	master.subnetAllocator.ReleaseNetwork(ipnet)
```

Comment:
deleteNode() is called from watchNodes() and this will break the normal case when the node is deleted. 
Also didn't like the fact that addNode() assigns the subnet but the deleteNode() doesn't revoke the subnet.